resolwe_bio/processes/expression/featureCounts_beta.py

```diff
@@ -0,0 +1,980 @@
+        annotation_file = inputs.annotation.output.annot.path
+        if (
+            inputs.annotation.output.source == "UCSC"
+            and inputs.annotation.type.startswith("data:annotation:gtf")
+        ):
+            with open(annotation_file, "r") as infile:
+                filedata = infile.read()
+
+            # Replace the missing gene_ids
+            annot_data = filedata.replace('gene_id "";', 'gene_id "unknown";')
+
+            # Write the output file
+            annotation_file = "annotation_modified.gtf"
+            with open(annotation_file, "w") as outfile:
+                outfile.write(annot_data)
+
+        fc_output = "featureCounts_rc.txt"
+
+        # Prepare featureCounts inputs
+        args = [
+            "-a",
+            annotation_file,
+            "-o",
+            fc_output,
+            "-F",
+            "GTF",
+            "-t",
+            inputs.feature_class,
+            "-g",
+            inputs.id_attribute,
+            "--minOverlap",
+            inputs.overlap.min_overlap,
+            "--fracOverlap",
+            inputs.overlap.frac_overlap,
+            "--fracOverlapFeature",
+            inputs.overlap.frac_overlap_feature,
+            "--readExtension5",
+            inputs.overlap.read_extension_5,
+            "--readExtension3",
+            inputs.overlap.read_extension_3,
+            "-Q",
+            inputs.read_filtering.min_mqs,
+            "--maxMOp",
+            inputs.miscellaneous.max_mop,
+            "-s",
+            strandedness,
+            "-T",
+            self.requirements.resources.cores,
+        ]
+
+        if inputs.general.count_features:
+            args.append("-f")
+
+        if inputs.overlap.allow_multi_overlap:
+            args.append("-O")
+
+        if inputs.overlap.largest_overlap:
+            args.append("--largestOverlap")
+
+        if inputs.overlap.read_to_pos:
+            args.extend(["--read2pos", inputs.overlap.read_to_pos])
+
+        if inputs.general.count_multi_mapping_reads:
+            args.append("-M")
+
+        if inputs.general.fraction:
+            args.append("--fraction")
+
+        if inputs.read_filtering.split_only:
+            args.append("--countSplitAlignmentsOnl")
+
+        if inputs.read_filtering.non_split_only:
+            args.append("--countNonSplitAlignmentsOnly")
+
+        if inputs.read_filtering.primary:
+            args.append("--primary")
+
+        if inputs.read_filtering.ignore_dup:
+            args.append("--ignoreDup")
+
+        if inputs.exon_exon_junctions.junc_counts:
+            args.append("-J")
+
+        if inputs.exon_exon_junctions.junc_counts and inputs.exon_exon_junctions.genome:
+            args.extend(["-G", inputs.exon_exon_junctions.genome.path])
+
+        if inputs.general.by_read_group:
+            # Check if @RG is in header of the BAM file
+            (Cmd["samtools"]["view", "-H"][bam_file] > "header.sam")()
```

Comment:
`-o` option could be used to avoid piping and to improve error handling. If the output is a `.txt` file the check can be done in Python.